app/javascript/components/users/SignInPage.spec.tsx

```diff
@@ -68,22 +68,31 @@ async function locationAssign(input: (locationAssignSpy: jest.SpyInstance<void,
 describe('Links', () => {
 	it('forgot password Link goes to correct path', async () => {
 		expect.hasAssertions();
-		locationAssign(async (locationAssignSpy: jest.SpyInstance<void, [url: string]>) => {
+		await locationAssign(async (locationAssignSpy: jest.SpyInstance<void, [url: string]>) => {
 			const { getByText } = render(<Wrapper><SignInPage redirectUrl={'redirectUrl'} /></Wrapper>);
 			await waitFor(() => {
-				const forgotPassword = getByText('Forgot Password?');
-				fireEvent.click(forgotPassword);
-				expect(locationAssignSpy).toHaveBeenCalledWith('/users/password/new');
+				expect(getByText('Forgot Password?')).toBeInTheDocument();
+			});
+			act(() => {
+				fireEvent.click(getByText('Forgot Password?'));
 			});
+
+			await waitFor(() => {
+				expect(locationAssignSpy).toHaveBeenCalledWith('/users/password/new');
+			})
 		});
 	});
-	it('terms & privacy Link has correct path', () => {
+	it('terms & privacy Link has correct path', async () => {
 		expect.hasAssertions();
-		locationAssign(async (locationAssignSpy: jest.SpyInstance<void, [url: string]>) => {
+		await locationAssign(async (locationAssignSpy: jest.SpyInstance<void, [url: string]>) => {
 			const { getByTestId } = render(<Wrapper><SignInPage redirectUrl={'redirectUrl'} /></Wrapper>);
 			await waitFor(() => {
-				const terms = getByTestId('termsTest');
-				fireEvent.click(terms);
+				expect(getByTestId('termsTest')).toBeInTheDocument();
+			});
+			act(() => {
+				fireEvent.click(getByTestId('termsTest'));
+			});
+			await waitFor(() => {
```

Comment:
These are still failing because, for some reason, the user is already logged in. I think we need to migrate the setup from SignInComponent.spec.tsx.